src/cljs/owlet_ui/events.cljs

```diff
@@ -132,25 +132,30 @@
 
     (rf/dispatch [:process-activity-metadata metadata])
 
-    ; Obtains the URL for each preview image, and adds a :url field next to
-    ; its :id field in [:activities :fields :preview :sys] map.
-
-    (let [url-for-id                                        ; Maps preview image IDs to associated URLs.
-          (->> (get-in entries [:includes :Asset])
-               (map (juxt (comp :id :sys)
-                          (comp :url :file :fields)))
-               (into {}))
+    ; Obtains the URL, width, and height for each image (asset), and
+    ;  - for each *preview* image, adds a :url field next to its :id field in [:activities :fields :preview :sys] map.
+    ;  - for each *gallery* image, adds a map to :image-gallery-items field in [:activities :fields] map.
+
+    (let [image-by-id                                       ; Maps image IDs to associated URL, width, and height.
+           (zipmap
+             (->> (get-in entries [:includes :Asset])
```

Comment:
@eemshi great work here 👏 
one small refactor suggestion, looks like we can pull asset into its own binding since its used twice
```
  (let [asset (get-in entries [:includes :Asset])
         image-by-id
          (zipmap
            (->> asset
                 (mapv (comp :id :sys)))
            (->> asset
                 (mapv #(hash-map :url (get-in % [:fields :file :url])
                                  :w (get-in % [:fields :file :details :image :width])
                                  :h (get-in % [:fields :file :details :image :height])))))
...
```
no biggie though